includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -270,6 +301,34 @@ public static function get_tax_rates_for_item( WC_Order_Item $item, WC_Subscript
 		return $tax_rates;
 	}
 
+	/**
+	 * Prepares discount data used to create a WCPay subscription.
+	 *
+	 * @param WC_Subscription $subscription The WC subscription used to create the subscription on server.
+	 * @param bool            $parent       Whether to get data from subscription parent.
```

Comment:
This is to accommodate a sign-up fee coupon. This type of coupon only appears on the parent order, but not on the subscription. Since we are using this method to also validate and repair discount data in our invoice service, we need to distinguish between the WC parent order for new subscriptions and the WC subscription for renewals.

I hope this is clear. Let me know if you still have questions.